### PR TITLE
reef: ceph.spec.in: add support for openEuler OS

### PR DESCRIPTION
**Pull request by liuqinfei**
backport: [PR50301](https://github.com/ceph/ceph/pull/50301)
The ceph main branch daily build on openEuler has been built. https://github.com/openeuler-mirror/ceph-daily-build The verification includes:
    - build ceph from source
    - run unit test
    - build RPM package

Signed-off-by: liuqinfei [lucas.liuqinfei@huawei.com](mailto:lucas.liuqinfei@huawei.com)

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/install/get-packages.rst [hunk 1]
```diff
@@ -251,6 +251,17 @@ openSUSE Tumbleweed
 The newest major release of Ceph is already available through the normal Tumbleweed repositories.
 There's no need to add another package repository manually.
 
+openEuler
+^^^^^^^^^
+
+There are two major versions supported in normal openEuler repositories. They are ceph 12.2.8 in openEuler-20.03-LTS series and ceph 16.2.7 in openEuler-22.03-LTS series. There’s no need to add another package repository manually.
+You can install ceph just by executing the following:
```
Comment (anthonyeleven) [inline doc/install/get-packages.rst:258]: s/ceph just/Ceph/

File: doc/install/get-packages.rst [hunk 2]
```diff
@@ -251,6 +251,17 @@ openSUSE Tumbleweed
 The newest major release of Ceph is already available through the normal Tumbleweed repositories.
 There's no need to add another package repository manually.
 
+openEuler
+^^^^^^^^^
+
+There are two major versions supported in normal openEuler repositories. They are ceph 12.2.8 in openEuler-20.03-LTS series and ceph 16.2.7 in openEuler-22.03-LTS series. There’s no need to add another package repository manually.
```
Comment (anthonyeleven) [inline doc/install/get-packages.rst:257]: two Ceph releases
They are Ceph 12.2.8 in the 
and Ceph 16.2.7 in the

File: doc/install/get-packages.rst (hunk 1 above)
Comment (liuqinfei) [reply doc/install/get-packages.rst]: Thanks. updated.

File: doc/install/get-packages.rst (hunk 2 above)
Comment (liuqinfei) [reply doc/install/get-packages.rst]: Thanks. updated.

**Comment by Svelar** [comment]
jenkins test api

**Comment by Svelar** [comment]
jenkins test windows

**Comment by tchaikov** [comment]
please use `git cherry-pick -x` to cherry pick from main.

**Comment by liuqinfei** [comment]
> > please use `git cherry-pick -x` to cherry pick from main.
> 
> ... and since you are backporting to quincy and reef, please open a backport to squid as well.

Because this PR was merged to the main branch a long time ago. And then the S version was pulled out from the main branch. So the S version doesn't need a backport.

Since a new PR has been submitted [56361](https://github.com/ceph/ceph/pull/56361). Close this one. Thanks.

**Comment by idryomov** [comment]
> Because this PR was merged to the main branch a long time ago. And then the S version was pulled out from the main branch. So the S version doesn't need a backport.

Ah, right -- I got confused with the timeline.  My apologies.

**Comment by liuqinfei** [comment]
> > Because this PR was merged to the main branch a long time ago. And then the S version was pulled out from the main branch. So the S version doesn't need a backport.
> 
> Ah, right -- I got confused with the timeline. My apologies.

Never mind, I was confused myself too at first.

**Pull request closed without merging**